src/models/game_session.ts

```diff
@@ -104,6 +104,13 @@ export default class GameSession {
             this.connection.removeAllListeners();
         }
         this.stopGuessTimeout();
+        if (this.gameType === GameType.ELIMINATION) {
+            const eliminationScoreboard = this.scoreboard as EliminationScoreboard;
```

Comment:
Do we need this cast?

Reply:
We can make the `gameFinished` check for classic mode in here as well

Reply:
The cast was needed to call `eliminationScoreboard`'s `gameFinished` method instead of `Scoreboard`'s

Reply:
> We can make the `gameFinished` check for classic mode in here as well

To do this, we need to access `guildPreference` for `isGoalSet`, so we need to call `getGuildPreference`. However, to await this, I need to turn endRound into an asynchronous method. Is that okay?